src/components/Pager/Pager.js

```diff
@@ -0,0 +1,299 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import classnames from 'classnames';
+import { TooltipHover } from '../../index';
+
+class PagerListItem extends Component {
+  static propTypes = {
+    onClick: PropTypes.func,
+    onKeyUp: PropTypes.func,
+    currentPage: PropTypes.number,
+    selected: PropTypes.bool,
+    id: PropTypes.func,
+  };
+
+  render() {
+    const { selected, currentPage } = this.props;
+    const classes = classnames('bx--pager__page-item', {
+      'bx--pager-item--selected': selected,
+    });
+
+    return (
+      <button
+        onKeyUp={this.props.onKeyUp}
+        onClick={this.props.onClick}
+        onKeyDown={e => {
+          e.key === 'Enter' ? e.preventDefault() : '';
+        }}
+        className={classes}
+        type="button"
+        ref={this.props.id}
+      >
+        {currentPage}
+      </button>
+    );
+  }
+}
+
+export default class Pager extends Component {
+  constructor(props) {
+    super(props);
+    const { totalItems, initialPage, mid } = this.props;
+    const lower = [2, 3];
+    const higher = [totalItems - 2, totalItems - 1];
+    const isSmall = totalItems <= this.props.max;
+    const pagesArr = Array.from(new Array(totalItems), (val, index) => index + 1);
+
+    this.state = {
+      activePage: initialPage,
+      leftPageQueue: lower,
+      rightPageQueue: higher,
+      activeQueue: isSmall ? pagesArr : lower,
+      small: isSmall,
+      showLower: initialPage < mid,
+      showCenter: initialPage > mid,
+      showHigher: initialPage >= totalItems - 2,
+    };
+  }
+
+  static propTypes = {
+    className: PropTypes.string,
+    initialPage: PropTypes.number,
+    totalItems: PropTypes.number,
+    max: PropTypes.number,
```

Comment:
Does this relate solely to truncation? If so, again maybe we could name the prop around truncation here for clarity.

Reply:
👍